packages/in_app_purchase/in_app_purchase_ios/lib/src/store_kit_wrappers/enum_converters.dart

```diff
@@ -40,6 +41,10 @@ class SKTransactionStatusConverter
       case SKPaymentTransactionStateWrapper.restored:
         return PurchaseStatus.restored;
       case SKPaymentTransactionStateWrapper.failed:
+        if (error != null && (error.code == 2 || error.code == 15)) {
```

Comment:
It would be good to add a comment explaining the error codes and referring to the official Apple documentation. Maybe something like this:

```suggestion
        // According to the Apple documentation the error code "2" indicates
        // the user cancelled the payment (SKErrorPaymentCancelled) and error
        // code "15" indicates the cancellation of the overlay (SKErrorOverlayCancelled).
        // An overview of all error codes can be found at: https://developer.apple.com/documentation/storekit/skerrorcode?language=objc
        if (error != null && (error.code == 2 || error.code == 15)) {
```